### DISCUSSION TITLE
Better touchscreen support on linux version

### DISCUSSION THREAD

**Discussion by UlyssesZh** [ppy/osu#18757] [Ideas] [answered]
I have a touchscreen connected to my computer (OS: Ubuntu 20.04.4 LTS, Windowing System: X11). I wonder whether it can be used to play osu!, but it turns out that osu!lazer is not treating touch input properly. If high-precision mouse is enabled, touchscreen act like touchpad. If high-precision mouse is disabled, I can hit *click*s normally but I cannot drag the cursor so I cannot hit *slide*s normally. There are also other issues like #14645 (although I did not encounter this problem myself). In other words, osu!lazer should detect the player's input devices (tablet, mouse, or touchscreen) to provide the optimum playstyle correctly and automatically (or, provide clear options for the players and tell them to select their playstyle when they launch osu!lazer for the first time). Specifically, on touchscreen, I am expecting something like osu!droid (like what #4201 suggested) or something like osu!stream.

**Comment by frenzibyte** [comment, marked as answer]
Touch input on desktop platforms is not supported currently (tracked at https://github.com/ppy/osu-framework/issues/1514), and touch input for rulesets is also not supported properly (tracked at https://github.com/ppy/osu/issues/4201).
